packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -30,6 +30,22 @@ enum Orientation {
 /// If no [MediaQuery] is in scope then the [MediaQuery.of] method will throw an
 /// exception, unless the `nullOk` argument is set to true, in which case it
 /// returns null.
+///
+/// MediaQueryData includes two [EdgeInsets] values:
+/// [padding] and [viewInsets]. These
+/// values reflect the configuration of the device and are used by
+/// many top level widgets, like [SafeArea] and the cupertino and
+/// material scaffold widgets. The padding value defines areas that
+/// might not be completely visible, like the display "notch" on the
+/// iPhone X. The viewInsets value defines areas that aren't visible at
+/// all, typically because they're obscured by the device's keyboard.
+/// The viewInsets and padding values are independent, they're both
```

Comment:
While you're here... :D

Would you like to add another section describing their interplay? I have to go look it up and/or test it every time.

i.e. when the Android/iOS home bar (padding) and the keyboard (inset) are on top of each other, does the inset include the padding, is the padding just zero, does the inset only count the area on top of the padding etc etc. I can never remember since we didn't write it down.

Reply:
Might be worth saying too: if you make a custom widget and it consumes a padding/inset, you're responsible to make sure your child doesn't see it anymore. Otherwise, your child might decide to be funky.

Reply:
Oops, the second comment is in the wrong place.

Reply:
The MediaQueryData class doc covers this. I separated the point into its own paragraph to make it more obvious:
```
/// The viewInsets and padding values are independent, they're both
/// measured from the edges of the MediaQuery widget's bounds. The
/// bounds of the top level MediaQuery created by [WidgetsApp] are the
/// same as the window that contains the app.
```